### PR TITLE
Fixed Bug 1796083: Create Receiver page loads forever when Prometheus API request returns an error

### PR DESCRIPTION
**Pull request by dtaylor113**
Added reject method to coFetchJSON(`${alertManagerBaseURL}/api/v2/status/`).then(...)

**Now we see this after a few seconds:**

image

### CODE REVIEW

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx [hunk 1]
```diff
@@ -435,22 +435,29 @@ const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(({ obj,
...
+          } catch ({ message }) {
+            setLoadError({
+              message: `Error parsing Alertmanager config.original: ${message || 'invalid YAML'}`,
+            });
+          }
         }
-      }
-    });
+      },
+      (reason) => {
```
Comment (spadgett) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Use `coFetchJSON(...).then(...).catch(...)` instead

Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: fixed

Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Hmm...I could just do `.catch((e) => setLoadError(e));`, which would result in showing just the following on the page (after one hits the [Create Receiver] button):
image

Which would match what one sees in Monitoring -> Alerts | Silences:

image

idk, kinda feel like the error message `Error Loading Alertmanager Globals: Error loading /api/alertmanager/api/v2/status/: Error: Forbidden` is more descriptive.

File: frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx [hunk 2]
```diff
@@ -435,22 +435,27 @@ const ReceiverWrapper: React.FC<ReceiverFormsWrapperProps> = React.memo(({ obj,
...
+              message: `Error parsing Alertmanager config.original: ${message || 'invalid YAML'}`,
+            });
+          }
         }
-      }
-    });
+      })
+      .catch((e) => {
+        e.message = `Error loading ${alertManagerBaseURL}/api/v2/status/: ${e}`;
+        setLoadError(e);
```
Comment (spadgett) [inline frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: No reason to mutate `e`

```suggestion
        setLoadError({ message: `Error loading ${alertManagerBaseURL}/api/v2/status/: ${e.message}` });
```

Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Fixed.  I had it that way originally, but we lose the 'Error:' in '...Error: Forbidden', but I think it's ok:
image

Comment (dtaylor113) [reply frontend/public/components/monitoring/receiver-forms/alert-manager-receiver-forms.tsx]: Fixed

**Pull request merged**